File: Autodrawer/Autodrawer/src/Models/Canvas.cs
using System.Drawing;

namespace Autodrawer.Models;

public class Canvas
{
    private Point TopLeftCorner { get; }
    private Point BottomRightCorner { get; }
    private Point Center { get; }
    private Point StartingPoint { get; } // The cursor will start at this Point
    private Size ImageSize { get; }
    private Size CanvasSize { get; }

    public Canvas(Point topLeftCorner, Point bottomRightCorner, Size imageSize)
    {
        TopLeftCorner = topLeftCorner;
        BottomRightCorner = bottomRightCorner;
        ImageSize = imageSize;

        CanvasSize = new Size(BottomRightCorner.X - TopLeftCorner.X,BottomRightCorner.Y - TopLeftCorner.X); // TODO: Is this right? Compare first and second parameters
        Center = new Point((BottomRightCorner.X + TopLeftCorner.X) / 2,(BottomRightCorner.Y + TopLeftCorner.Y) / 2);
        StartingPoint = new Point((Center.X - imageSize.Width) / 2,(Center.Y - ImageSize.Height) / 2);
    }
}

File: src/models/Canvas.ts
import { Dimension } from "../types/Dimension";
import { Point } from "../types/Point";

export class Canvas {
    private readonly _topLeftCorner: Point
    private readonly _bottomRightCorner: Point
    private readonly _center: Point
    private readonly _startingPoint: Point
    private readonly _imageSize: Dimension
    private readonly _canvasSize: Dimension


    constructor(topLeftCorner: Point, bottomRightCorner: Point, imageSize: Dimension) {
        this._topLeftCorner = topLeftCorner;
        this._bottomRightCorner = bottomRightCorner;
        this._imageSize = imageSize;

        this._canvasSize = {
            width: this._bottomRightCorner.x - this._topLeftCorner.x,
            height: this._bottomRightCorner.y - this._topLeftCorner.x
        }

        this._center = {
            x: (this._bottomRightCorner.x + this._topLeftCorner.x) / 2,
            y: (this._bottomRightCorner.y + this._topLeftCorner.y) / 2
        }

        this._startingPoint = {
            x: (this._center.x - this._imageSize.width) / 2,
            y: (this._center.y - this._imageSize.height) / 2
        }
    }

    get topLeftCorner(): Point {
        return this._topLeftCorner;
    }

    get bottomRightCorner(): Point {
        return this._bottomRightCorner;
    }

    get center(): Point {
        return this._center;
    }

    get startingPoint(): Point {
        return this._startingPoint;
    }

    get imageSize(): Dimension {
        return this._imageSize;
    }

    get canvasSize(): Dimension {
        return this._canvasSize;
    }
}